### PR TITLE
Adds bluespace miner

### PR DESCRIPTION
**Pull request by karmaisblackandbluepilled**
Adds the bluespace miner. Automatically generates resources slowly while consuming power. Needs bluespace crystals, gold and uranium to be made.
originally made by [Toxici11i](https://github.com/Toxici11i) over on hippie
help why is tgstation.dme there a aa aa

### CODE REVIEW

**Comment by AcapitalA** [comment]
"It's not a question IF the miners die, the question is WHEN."
Based commit name btw.

**Comment by TheChosenEvilOne** [comment]
Not too sure on this due to #30, but it might take a while to get ready.

File: code/modules/mining/bluespace.dm [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	desc = "A machine that uses the magic of Bluespace to slowly generate materials and add them to a linked ore silo."
+	icon = 'icons/obj/machines/mining_machines.dmi'
+	icon_state = "stacker"
+	density = TRUE
+	circuit = /obj/item/circuitboard/machine/bluespace_miner
+	layer = BELOW_OBJ_LAYER
+	processing_flags = START_PROCESSING_ON_INIT
+	var/list/ores = list(/datum/material/iron = 600, /datum/material/plasma = 400,  /datum/material/silver = 400, /datum/material/gold = 250, /datum/material/titanium = 250, /datum/material/uranium = 250, /datum/material/bananium = 90, /datum/material/diamond = 90, /datum/material/bluespace = 90)
+	var/datum/component/remote_materials/materials
+	var/debugging = 0
```
Comment (TheChosenEvilOne) [inline code/modules/mining/bluespace.dm:42]: remove this

File: code/modules/mining/bluespace.dm [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+/obj/machinery/mineral/bluespace_miner/process()
+	if(!materials?.silo || materials?.on_hold())
+		return
+	var/datum/component/material_container/mat_container = materials.mat_container
+	if(!mat_container || panel_open || !powered())
+		return
+	var/datum/material/ore = pick(ores)
+	materials.mat_container.insert_amount_mat(ores[ore], ore)
+	if(debugging == 1)
+		materials.mat_container.insert_amount_mat(10000,/datum/material/iron)
```
Comment (TheChosenEvilOne) [inline code/modules/mining/bluespace.dm:79]: ditto

**Comment by TheChosenEvilOne** [comment]
oh well, this can be removed anyways once factorio mining is done.

**Pull request merged**